lib/Sema/CodeSynthesis.cpp

```diff
@@ -2504,6 +2504,18 @@ synthesizeAccessorBody(AbstractFunctionDecl *fn, void *) {
   llvm_unreachable("bad synthesized function kind");
 }
 
+static bool isNilOrTupleOfNils(Type type, ASTContext &ctx) {
```

Comment:
This is really "isOptionalOrTupleOfOptionals". You're looking at a type, not a value. You should get the ASTContext from the Type.